syntax = "proto3";

message Consignment {
    int32 id = 1;
    string name = 2;
    int32 weight = 3;
}

message ConsignmentVessel {
    int32 id = 1;
    int32 max_weight = 2;
}

message GetConsignmentRequest {
    int32 consignment_id = 1;
}

service Shipping {
	rpc GetConsignment(GetConsignmentRequest) returns (Consignment) {}
    rpc GetVesselForConsignment(Consignment) returns(ConsignmentVessel) {}
}
